hash_reader.go

```diff
@@ -133,43 +172,98 @@ func (hashReader *hashReader) TimeValue() (time.Time, error) {
 }
 
 func (hashReader *hashReader) StringValue() (string, error) {
-	panic("implement me")
 	return hashReader.ionReader.StringValue()
 }
 
 func (hashReader *hashReader) ByteValue() ([]byte, error) {
-	panic("implement me")
 	return hashReader.ionReader.ByteValue()
 }
 
 func (hashReader *hashReader) Sum(b []byte) []byte {
-	panic("implement me")
+	hashReader.hasher.digest()
+}
+
+func (hashReader *hashReader) traverse() error {
+	for hashReader.Next() {
+		switch hashReader.typeIon {
+		case ion.ListType:
+		case ion.SexpType:
+		case ion.StructType:
+			if !hashReader.IsNull() {
+				err := hashReader.StepIn()
+				if err != nil {
+					return err
+				}
+
+				err = hashReader.traverse()
+				if err != nil {
+					return err
+				}
+
+				err = hashReader.StepOut()
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
 }
 
 // The following implements HashValue interface.
 
 func (hashReader hashReader) getFieldName() string {
-	panic("implement me")
+	return hashReader.FieldName()
 }
 
 func (hashReader hashReader) getAnnotations() []string {
-	panic("implement me")
+	return hashReader.Annotations()
 }
 
 func (hashReader *hashReader) value() interface{} {
```

Comment:
lets update this signature to (interface{}, error)

Reply:
Done.